src/libvirtApi/storagePool.js

```diff
@@ -68,79 +68,75 @@ export function storagePoolDeactivate({ connectionName, objPath }) {
+    } catch (ex) {
+        console.warn('GET_STORAGE_POOL action failed for path', objPath, ex.toString());
+    }
 }
 
-export function storagePoolGetAll({
+export async function storagePoolGetAll({
     connectionName,
 }) {
-    return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'ListStoragePools', [0], { timeout, type: 'u' })
-            .then(objPaths => {
-                return Promise.all(objPaths[0].map(path => {
-                    return call(connectionName, path, 'org.freedesktop.DBus.Properties', 'Get', ['org.libvirt.StoragePool', 'Active'], { timeout, type: 'ss' })
-                            .then(active => {
-                                if (active[0].v)
-                                    return storagePoolRefresh({ connectionName, objPath: path });
-                                else
-                                    return storagePoolGet({ connectionName, id: path });
-                            });
-                }));
-            })
-            .catch(ex => {
-                console.warn('GET_ALL_STORAGE_POOLS action failed:', ex.toString());
-                return Promise.reject(ex);
-            });
+    try {
+        const [objPaths] = await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'ListStoragePools', [0],
+                                      { timeout, type: 'u' });
+        return await Promise.all(objPaths.map(async path => {
+            const [active] = await call(connectionName, path, 'org.freedesktop.DBus.Properties', 'Get',
+                                        ['org.libvirt.StoragePool', 'Active'], { timeout, type: 'ss' });
+            if (active.v)
+                return storagePoolRefresh({ connectionName, objPath: path });
+            else
+                return storagePoolGet({ connectionName, id: path });
+        }));
+    } catch (ex) {
+        console.warn('GET_ALL_STORAGE_POOLS action failed:', ex.toString());
+        throw ex;
+    }
 }
 
-export function storagePoolGetCapabilities({ connectionName }) {
-    // TODO: replace with D-Bus API once available https://bugzilla.redhat.com/show_bug.cgi?id=1986321
+export async function storagePoolGetCapabilities({ connectionName }) {
+    // TODO: replace with D-Bus API once available https://issues.redhat.com/browse/RHEL-7045
     const opts = { err: "message", environ: ['LC_ALL=C.UTF-8'] };
     if (connectionName === 'system')
         opts.superuser = 'try';
 
-    return cockpit.spawn(
-        ["virsh", "-c", "qemu:///" + connectionName, "pool-capabilities"],
-        opts
-    ).then(poolCapabilities => parsePoolCapabilities(poolCapabilities), ex => {
+    try {
+        const poolCapabilities = await cockpit.spawn(
+            ["virsh", "-c", "qemu:///" + connectionName, "pool-capabilities"],
+            opts);
+        return parsePoolCapabilities(poolCapabilities);
+    } catch (ex) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-075343-fbf2cd7c-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)